src/ngrx/1-why-ngrx/why-ngrx.md

```diff
@@ -5,6 +5,103 @@
 
 @body
 
-TODO
+1. Actively Maintained
 
-- Checkout repo
\ No newline at end of file
+    NgRx is [actively maintained](https://github.com/ngrx/platform) and keeps evolving with Angular.
+
+2. Great documentation
+
+    [NgRx documentation](https://ngrx.io/docs) is great and @ngrx/eslint-plugin can help you follow NgRx’s best practices.
+
+    PRO TIP: if you use [@bitovi/eslint-config](https://github.com/bitovi/eslint-config), there’s no need to worry: NgRx recommended ESLint rules will be configured for you.
+
+3. Maintainability
+
+    While NgRx allegedly has a lot of boilerplate code, NgRx does help keep your code tidy!
+
+4. Performance
+
+    NgRx is carefully implemented with performance in mind, and it complements the Angular Ecosystem perfectly because of its Reactive approach.
+
+## Key Concepts
+
+Understand the key concepts of NgRx
+
+### Redux Pattern
+
+The Redux pattern manages the global state of an application and aims at code testability and predictability.
+
+Redux is known for its boilerplate code. At first, it might seem like a lot of effort to implement Redux, but in fact it is a tradeoff. Redux adds a layer of indirection in order to improve maintainability, testability and predictability of your application. The tradeoff pays off more often than not particularly if your application is complex, large and maintained by a large team.
+
+[Redux expects updates to the state to be done immutably](https://redux.js.org/tutorials/essentials/part-1-overview-concepts#immutability). Not only does Immutability help Redux implement patterns that increase testability and predictability, it also improves performance, as change detection is achieved with less effort.
+
+#### Store
+
+A Store is an object that wraps the state. The Store also provides methods to update the state and listen to state changes.
+
+#### Actions
+
+Actions are objects that represent individual events in your application.
+
+Actions have two properties:
+
+1. `type` describes the event.
+2. `payload` (optional) holds additional information.
+
+Actions are necessary to update the state.
+
+#### Reducers
+
+To put it simply, reducers are pure functions that take the state and an action as arguments, and return a new state. In other words, Reducers are functions that, given an action, know how to update the state.
+
+Reducers functions must be `pure`:
+
+- do not depend on any information other than the action and current state;
+- do not perform side-effects, like mutating the current state. (side-effects are done through effects)
```

Comment:
in this part we are talking about Redux, so I think we could change to `(in NgRx, side-effects are done through effects)` (or handled by effects)